### PR TITLE
PT-167996886 A proper ecverify

### PR DESCRIPTION
**Pull request by hanssv**
Needs https://github.com/aeternity/aebytecode/pull/70
and https://github.com/aeternity/aesophia/pull/128

### CODE REVIEW

**Comment by hanssv** [comment]
Docs is currently being written.

File: Makefile [hunk 1]
```diff
@@ -292,7 +292,7 @@ smoke-test: docker smoke-test-run
 
 smoke-test-run: KIND=system_test
 smoke-test-run: internal-build
-	@$(REBAR) as $(KIND) do ct $(ST_CT_DIR) $(ST_CT_FLAGS) --suite=aest_sync_SUITE,aest_commands_SUITE,aest_peers_SUITE
+	@$(REBAR) as $(KIND) do upgrade, ct $(ST_CT_DIR) $(ST_CT_FLAGS) --suite=aest_sync_SUITE,aest_commands_SUITE,aest_peers_SUITE
```
Comment (dincho) [inline Makefile]: Isn't this `upgrade` workaround the lock file?

Comment (tolbrino) [reply Makefile]: The lock file is only used by the deps in the `default` profile. All additional deps defined in other profiles are not locked. Therefore, this upgrade ensure that they are updated to the defined reference.

Comment (dincho) [reply Makefile]: Interesting tool :)

Comment (hanssv) [reply Makefile]: @dincho there is no lock file for `system_test`?

Comment (hanssv) [reply Makefile]: And except that the update does not seem to help... Interesting tool indeed :-)

**Pull request merged**